arrow/arrow_test.go

```diff
@@ -61,3 +65,240 @@ func TestSum_Uint64_Empty(t *testing.T) {
 		t.Errorf("unexpected sum: %v != %v", got, want)
 	}
 }
+
+func TestSlice_Int64(t *testing.T) {
+	alloc := &mem.Allocator{}
+	arrowArray := arrow.NewInt([]int64{-5, -4, -3, -2, -1, 0, 1, 2, 3, 4}, alloc)
+	defer arrowArray.Release()
+
+	testcases := []struct {
+		slice *array.Int64
+		want  []int64
+	}{
+		{
+			slice: arrow.IntSlice(arrowArray, 0, 0),
+			want:  []int64{},
+		},
+		{
+			slice: arrow.IntSlice(arrowArray, 0, 5),
+			want:  []int64{-5, -4, -3, -2, -1},
+		},
+		{
+			slice: arrow.IntSlice(arrowArray, 0, 10),
+			want:  []int64{-5, -4, -3, -2, -1, 0, 1, 2, 3, 4},
+		},
+		{
+			slice: arrow.IntSlice(arrowArray, 5, 10),
+			want:  []int64{0, 1, 2, 3, 4},
+		},
+		{
+			slice: arrow.IntSlice(arrowArray, 10, 10),
+			want:  []int64{},
+		},
+		{
+			slice: arrow.IntSlice(arrowArray, 2, 7),
+			want:  []int64{-3, -2, -1, 0, 1},
+		},
```

Comment:
You might provide a test cases for invalid parameters, such as when the start index is less than the end index, or for indexing past then end of an array.

Reply:
Thanks for mentioning this. I forgot to add test cases for this to the arrow PR. Now there are test cases for **creating** a slice with invalid indices as well as test cases for **accessing** a slice using invalid indices.